### PR TITLE
Create volume out of the bytes data dir and change the default

### PR DESCRIPTION
**Pull request by Donnype**
## Changes
This builds rocky as root to have write permissions in the locale directory.

## Extra instructions for others
_This section may be skipped or omitted. Uncomment and answer the below questions if relevant._

## Checklist for author(s):
- [ ] All the commits in this PR are properly PGP-signed and verified;
- [ ] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [ ] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [ ] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [ ] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [ ] I have performed a self-review of my own code;
- [ ] I have commented my code, particularly in hard-to-understand areas;
- [ ] I have made corresponding changes to the documentation, if necessary;
- [ ] I have written unit, integration, and end-to-end tests for the change that I made;

If a non-trivial PR:
- [ ] This PR is part of a milestone and has appropriate labels;
- [ ] This PR is properly linked to the project board (either directly or via an issue);
- [ ] I have added screenshots or some other proof that my code does what it is supposed to do;

```
## Checklist for functional reviewer(s):
- [ ] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [ ] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [ ] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._
```

```
## Checklist for code reviewer(s):
- [ ] The code passes the CI tests and linters;
- [ ] The code does not bypass authentication or security mechanisms;
- [ ] The code does not introduce any dependency on a library that has not been properly vetted;
- [ ] The code does not violate Model-View-Template and our other architectural principles;
- [ ] The code contains docstrings, comments, and documentation where needed;
- [ ] The code prioritizes readability over performance where appropriate;
- [ ] The code conforms to our agreed coding standards.
```

### CODE REVIEW

File: bytes/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ WORKDIR /app/bytes
 ENV PATH=/home/bytes/.local/bin:${PATH}
 
 RUN mkdir /data && chown bytes: /data
+RUN mkdir ./bytes-data && chown bytes: ./bytes-data
```
Comment (ammar92) [inline bytes/Dockerfile:18]: You can actually set the file mode directly using `mkdir`s `--mode` option

**Review by ammar92 — COMMENTED** [review]
One small remark, but consider it to be approved

File: bytes/Dockerfile (hunk 1 above)
Comment (dekkers) [reply bytes/Dockerfile:18]: I don't understand why an extra data directory is created instead of just using /data? I think we should add a bytes-data volume and mount in on /data in the same way as what we do in the docker-compose.release-example.yml. Then we also don't need to do special things to make sure the directory also exists in the mounted volume and we don't pollute the source repository with data. 

@ammar92 The owner of the directory is being set however, not the mode.

File: bytes/bytes/raw/file_raw_repository.py [hunk 2]
```diff
@@ -21,7 +21,6 @@ def create_raw_repository(settings: Settings) -> RawRepository:
 
 class FileRawRepository(RawRepository):
     UUID_INDEX = 3  # To reduce the number of subdirectories based on the uuid, see self._index()
-    _ENCODING = "utf-8"
```
Comment (dekkers) [inline bytes/bytes/raw/file_raw_repository.py:24]: What is the purpose of this change?

File: rocky/Makefile [hunk 3]
```diff
@@ -9,7 +9,7 @@ export COMPOSE_DOCKER_CLI_BUILD=1
 build: build-rocky build-rocky-frontend
 
 build-rocky:
-	docker-compose run --rm rocky make build-rocky-native
+	docker-compose run -u root --rm rocky make build-rocky-native  # root for write permissions compilemessages
```
Comment (dekkers) [inline rocky/Makefile:12]: Compilemessages is already run in the Dockerfile. Is there any reason to run it again here or can we just remove it from the build-rocky-native step?

**Review by dekkers — CHANGES_REQUESTED** [review]

File: bytes/bytes/raw/file_raw_repository.py (hunk 2 above)
Comment (Donnype) [reply bytes/bytes/raw/file_raw_repository.py:24]: Discovered during looking into these issues that this is not used anywhere

File: rocky/Makefile (hunk 3 above)
Comment (Donnype) [reply rocky/Makefile:12]: The issue was the mount of the source code that removes the compiled messages when they are not present on the host (when debugging an install). Now that I am looking into the container file permissions, I realize that this might actually had to do with the host files being mounted may have had stricter permissions. I will have a look.

File: bytes/Dockerfile (hunk 1 above)
Comment (Donnype) [reply bytes/Dockerfile:18]: @dekkers This is partly a debugging artifact for the particular issue we were solving, but it got me thinking: what if users did not use the `docker-compose-release-example.yml` and still have the default value of `./bytes-data`. Then the build assumes you will run your containers with specific environment variables, which seemed a bit strange unless we explicitly set `ENV BYTES_DATA_DIR=/data` by default.

But given that this data residing in the source repository is mainly a legacy setup, I agree it is a lot nicer to add a bytes-data volume. Then we could add a command that copies all data from the previous default `./bytes-data` to this volume in the case people want to migrate the development docker-compose file?

`docker-compose run --rm --no-deps --entrypoint /bin/sh bytes -c "cp -r ./bytes-data/* /data"`

File: bytes/bytes/raw/file_raw_repository.py (hunk 2 above)
Comment (underdarknl) [reply bytes/bytes/raw/file_raw_repository.py:24]: Raw files are also not guaranteed to be text, right?

File: bytes/Dockerfile (hunk 1 above)
Comment (dekkers) [reply bytes/Dockerfile]: I am not sure whether we really need to provide migrations for this. It's a development setup, so there should only be test data in there?

I agree that we should also change the default value to `/data`.

**Comment by Donnype** [comment]
> Styling with Markdown is supportedCommentCancel
> Resolve conversation


This still shows some unexpected behavior, still looking into this

File: bytes/Dockerfile (hunk 1 above)
Comment (Donnype) [reply bytes/Dockerfile]: True, although it wouldn't hurt to add a line or two about it in the release notes of course

**Comment by Donnype** [comment]
@dekkers It turned out the issue was cause by Docker being run in rootless mode, causing files mounted from the host not to show the same UID and GID as the host but always having root permissions. I changed it back, but we might have to add either some text to the documentation or provide workaround for this?

**Review by dekkers — APPROVED** [review]
